ingestion/ingestion_server/ingestion_file_upload_rest_api.py

```diff
@@ -159,82 +159,101 @@ def file_upload():
+                    key_b64 = key_info["response"]["key"]
+                    logger.debug("key_id :%s", key_info["response"]["key_id"])
+                    key_iv_b64 = key_info["response"]["iv"]
+                    # Decode key material we got from KS
+                    # iv_decoded = base64.b64decode(key_iv_b64)
+                    iv_decoded = key_iv_b64
+                    logger.debug("decoded iv amd length: %s %s", iv_decoded, len(iv_decoded))
+                    # key_decoded = base64.b64decode(key_b64)
+                    key_decoded = key_b64
+                    logger.debug("decoded key and length: %s %s", key_decoded, len(key_decoded))
+                    success, file_size, message = EncryptionApi.encrypt_uploaded_file_and_save(file_obj, enc_file_path,
+                                                                                               key_decoded,
+                                                                                               iv_decoded)
+                    if not success:
+                        # TODO if something goes wrong we need to delete copy of file.
+                        raise Exception(message)
+
+                    asset_dict["file_size"] = file_size
+                    base64_file_path = enc_file_path + ".b64"
+                    success, message = EncryptionApi.write_base64_file_from_file(enc_file_path, base64_file_path)
+                    if not success:
+                        raise Exception(message)
+
+                    b64_file_digest, message = EncryptionApi.create_sha256_from_file(enc_file_path)
+                    if not b64_file_digest:
+                        raise Exception(message)
+
+                    metadata_json, metadata_dict = ContainerApi.create_meta_v2(asset_dict,
+                                                                               creator_domain="www.magen.io",
+                                                                               iv=iv_decoded,
+                                                                               enc_asset_hash=b64_file_digest.hexdigest())
+                    metadata_b64 = ContainerApi.b64encode_meta_v2(metadata_json)
+                    metadata_b64_str = metadata_b64.decode("utf-8")
+                    html_container_path = dst_file_path + ".html"
+
+                    if not ContainerApi.create_html_file_container_from_file(metadata_dict, metadata_b64_str,
+                                                                             base64_file_path, html_container_path):
+                        raise Exception("Failed to create container: {}".format(dst_file_path))
+                else:
+                    raise Exception("Key Server problem")
+
+            # GridFS file storage
+            db_core = MainDb.get_core_db_instance()
+            # TODO bucket name, owner and group should be based on user login such as email
+            if ext == 'pub':
```

Comment:
This does not look right. There is already a "if ext != 'pub'" above.  Here we are testing to see if it is equal to pub instead of using "else".